src/main/java/com/example/hashcartapp/service/AdvertisementService.java

```diff
@@ -0,0 +1,121 @@
+package com.example.hashcartapp.service;
+
+import com.example.hashcartapp.dto.AdvertisementDTO;
+import com.example.hashcartapp.entities.Advertisement;
+import com.example.hashcartapp.repository.AdvertisementRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.hibernate.internal.CoreLogging.logger;
+
+@Service
+public class AdvertisementService {
+
+    @Autowired
+    AdvertisementRepository advertisementRepository;
+
+    public List<AdvertisementDTO> getAllAdvertisements(){
+        return advertisementRepository.findAll()
+                .stream()
+                .map(this::convertEntityToDTO)
+                .collect(Collectors.toList());
+    }
+
+    public AdvertisementDTO getAdvertisementById(Long advertisementId) {
+        AdvertisementDTO advertisementToReturn = null;
+
+        try {
+            Optional<Advertisement> advertisementFound = advertisementRepository.findById(advertisementId);
+
+            if (advertisementFound.isPresent()) {
+                advertisementToReturn = advertisementFound.map(this::convertEntityToDTO).get();
+            }
+
+        } catch (NoSuchElementException e) {
+            logger("No advertisement found in the table having this id");
+            e.printStackTrace();
+        } catch (IllegalArgumentException e1) {
+            e1.printStackTrace();
+        } catch (Exception e2) {
+            e2.printStackTrace();
+        }
+        return advertisementToReturn;
+    }
+
+
+    public Advertisement saveAdvertisement(Advertisement advertisement){
+        return advertisementRepository.save(advertisement);
+    }
+
+
+    public void deleteAdvertisement(Long advertisementId){
+        advertisementRepository.deleteById(advertisementId);
+    }
```

Comment:
usually in most enterprise applications, we only soft delete ( like using a deleted flag)